beacon_node/lighthouse_network/src/service/mod.rs

```diff
@@ -353,34 +353,71 @@ impl<AppReqId: ReqId, TSpec: EthSpec> Network<AppReqId, TSpec> {
+                self.0.spawn(f, "libp2p");
             }
+        }
 
-            // sets up the libp2p connection limits
-
-            (
-                SwarmBuilder::with_executor(
-                    transport,
-                    behaviour,
-                    local_peer_id,
-                    Executor(executor),
-                )
-                .notify_handler_buffer_size(std::num::NonZeroUsize::new(7).expect("Not zero"))
-                .per_connection_event_buffer_size(4)
-                .build(),
-                bandwidth,
-            )
-        };
+        let (swarm_builder, bandwidth) =
+            libp2p::SwarmBuilder::with_existing_identity(local_keypair.clone())
+                .with_tokio()
+                // Using `with_other_transport` in order to support mplex. In case mplex would not be needed, the below would be as simple as:
+                // ```
+                // .with_tcp()
+                // .with_noise()?
+                // ```
+                .with_other_transport(|_| {
+                    let tcp = libp2p::tcp::tokio::Transport::new(
+                        libp2p::tcp::Config::default().nodelay(true),
+                    );
+                    let transport = libp2p::dns::TokioDnsConfig::system(tcp)?;
+
+                    // TODO If you want to support websocket, you can chain a `.with_websocket` before the `.with_bandwidth_logging`.
+                    //
+                    // #[cfg(feature = "libp2p-websocket")]
+                    // let transport = {
+                    //     let trans_clone = transport.clone();
+                    //     transport.or_transport(libp2p::websocket::WsConfig::new(trans_clone))
+                    // };
+
+                    // mplex config
+                    let mut mplex_config = libp2p_mplex::MplexConfig::new();
+                    mplex_config.set_max_buffer_size(256);
+                    mplex_config.set_max_buffer_behaviour(libp2p_mplex::MaxBufferBehaviour::Block);
+
+                    // yamux config
+                    let mut yamux_config = libp2p::yamux::Config::default();
+                    yamux_config.set_window_update_mode(libp2p::yamux::WindowUpdateMode::on_read());
+
+                    Ok(transport
+                        .upgrade(libp2p::core::upgrade::Version::V1)
+                        .authenticate(
+                            libp2p::noise::Config::new(&local_keypair)
+                                .expect("signing can fail only once during starting a node"),
+                        )
+                        .multiplex(libp2p::core::upgrade::SelectUpgrade::new(
+                            yamux_config,
+                            mplex_config,
+                        ))
+                        // TODO: Enough to just streammuxerbox
+                        .boxed())
+                })
+                // TODO: Handle
+                .unwrap()
```

Comment:
Adding QUIC as a transport would be as easy as:

```suggestion
                .unwrap()
                .with_quic()
```